jQuery.tubeplayer.js

```diff
@@ -267,15 +267,58 @@
 		if (arguments.length === 0 || type === "object") {
 
 			return $this.each(function() {
+				$this = $(this);
+				// check if the current element is an iframe and if so replace it with a div
+				// Mihai Ionut Vilcu - 26/Oct/2013
+				if($this.prop('tagName') == "IFRAME") {
+					// make sure we have a valid YT url
+		            var regExp = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#\&\?]*).*/;
+
+		            var match = $this.attr('src').match(regExp);
+
+		            if (match&&match[7].length==11) { // we have the id
+
+		            	var settings = {initialVideo : match[7]},
+		            		class_name = $this.attr('class'),
+		            		ids = $this.attr('id');
+
+		            	// check for extra settings
+						if($this.attr('width'))
+							settings.width = $this.attr('width');
+
+						if($this.attr('height'))
+							settings.height = $this.attr('height');
+
+						if($this.attr('allowfullscreen') != undefined)
+							settings.allowFullScreen = "true";
+						else
+							settings.allowFullScreen = false;
 
-				TP.init($(this), input);
+		            	var newDiv = $("<div></div>").attr({
+		            		'class' : class_name ? class_name : '',
+		            		'id' : ids ? ids : '',
+		            	});
 
+		            	$this.replaceWith(newDiv);
+
+		            	var new_input = $.extend({}, defaults, settings, input);
+
+						TP.init(newDiv, new_input);
+
+		            }
+
+				} else {
+
+					TP.init($(this), input);
+				}
 			});
 
 		} else if (type === "string") {
 
-			return $this.triggerHandler(input + TUBEPLAYER, (typeof xtra !== 'undefined' ? xtra : null));
-		
+			return $this.each(function(index, el) { // handle multiple elemnts
```

Comment:
Oops. This shouldn't be here (and I shouldn't have merged it in without checking) - when this is being invoked, we want the response from the command execution. e.g. `.tubeplayer("volume")` should return an int representing the volume, not the elements.